content_sync/pipelines/concourse.py

```diff
@@ -249,3 +259,98 @@ def get_build_status(self, build_id: int):
+        return f"/api/v1/teams/{settings.CONCOURSE_TEAM}/pipelines/ocw-theme-assets/jobs/{job_name}/builds?vars={self.instance_vars}"
+
+    def _make_pipeline_config_url(self):
+        """Make URL for fetching pipeline info"""
+        return f"/api/v1/teams/{settings.CONCOURSE_TEAM}/pipelines/ocw-theme-assets/config?vars={self.instance_vars}"
+
+    def _make_job_url(self, job_name: str):
+        """Make URL for fetching job info"""
+        return f"/api/v1/teams/{settings.CONCOURSE_TEAM}/pipelines/ocw-theme-assets/jobs/{job_name}?vars={self.instance_vars}"
+
+    def _make_pipeline_unpause_url(self):
+        """Make URL for unpausing a pipeline"""
+        return f"/api/v1/teams/{settings.CONCOURSE_TEAM}/pipelines/ocw-theme-assets/unpause?vars={self.instance_vars}"
+
+    def upsert_theme_assets_pipeline(self):
+        """Upsert the theme assets pipeline"""
+        with open(
+            os.path.join(
+                os.path.dirname(__file__),
+                "definitions/concourse/theme-assets-pipeline.yml",
+            )
+        ) as pipeline_config_file:
+            config_str = (
+                pipeline_config_file.read()
+                .replace("((ocw-hugo-themes-uri))", OCW_HUGO_THEMES_GIT)
+                .replace("((ocw-hugo-themes-branch))", settings.GITHUB_WEBHOOK_BRANCH)
+                .replace("((search-api-url))", settings.SEARCH_API_URL)
+                .replace("((ocw-bucket-draft))", settings.AWS_PREVIEW_BUCKET_NAME)
+                .replace("((ocw-bucket-live))", settings.AWS_PUBLISH_BUCKET_NAME)
+            )
+            config = json.dumps(yaml.load(config_str, Loader=yaml.SafeLoader))
+            log.debug(config)
+            # Try to get the version of the pipeline if it already exists, because it will be
+            # necessary to update an existing pipeline.
+            url_path = self._make_pipeline_config_url()
+            try:
+                _, headers = self.api.get_with_headers(url_path)
+                version_headers = {
+                    "X-Concourse-Config-Version": headers["X-Concourse-Config-Version"]
+                }
+            except HTTPError:
+                version_headers = None
+            self.api.put_with_headers(url_path, data=config, headers=version_headers)
+
+    def trigger_pipeline_build(self) -> int:
+        """Trigger a pipeline build"""
+        pipeline_info = self.api.get(self._make_pipeline_config_url())
+        job_name = pipeline_info["config"]["jobs"][0]["name"]
+        return self.api.post(self._make_builds_url(job_name))["id"]
+
+    def unpause_pipeline(self):
+        """Unpause the pipeline"""
+        self.api.put(self._make_pipeline_unpause_url())
+
+    def get_build_status(self, build_id: int):
+        """Retrieve the status of the build"""
+        return self.api.get_build(build_id)["status"]
+
+    def abort_build(self, build_id: int):
+        """Abort a build"""
+        return self.api.abort_build(build_id)
```

Comment:
Same as comment above (put in the base pipeline to avoid dupe or nearly identical code)